app/controllers/claims_controller.rb

```diff
@@ -25,6 +25,10 @@ def show
 
     if params[:slug] == "teacher-detail"
       save_and_set_teacher_id_user_info
+    elsif params[:slug] == "qualification-details"
+      redirect_slug = next_slug
```

Comment:
This isn't possible because `Dqt::RetrieveClaimQualificationsData` changes the Claim object and so the subsequent call to `next_slug` could delete the current `qualification-details` slug from the slug sequence depending on what changed. When the current slug is missing, `next_slug` reverts the user back to the `teacher-details` page. I think we should ideally move this service object call to an `update` action, which would probably have to be on the `teacher-details` slug, so the claim is not updated during the `show` action.